docs/specs/tsd-006-startup-error-behavior.md

```diff
@@ -0,0 +1,101 @@
+# Startup Error Behavior
+
+## Objective
+
+Unified, configurable behavior on retry-able startup errors.
+
+## Keywords
+
+inputs, outputs, startup, error, retry
+
+## Overview
+
+Many Telegraf plugins connect to an external service either on the same machine
+or via network. On automated startup of Telegraf (e.g. via service) there is no
+guarantee that those services are fully started yet, especially when they reside
+on a remote host. More and more plugins implement mechanisms to retry reaching
+their related service if they failed to do so on startup.
+
+This specification intends to unify the naming of configuration-options, the
+values of those options and their semantic meaning. It describes the behavior
+for the different options on handling startup-errors.
+
+Startup-errors are all errors occurring in calls to `Start()`, for inputs and
+service-inputs, or `Connect()`, for outputs. The behaviors described below
+should only be applied in cases where the plugin *explicitly* states that an
+startup-error is *retry-able*. This comprises for example network errors
+indicating that the host or service is not reachable (yet) or external
+resources, like a machine or file, which are not yet available but might become
+available later. To indicate a retry-able startup-error the plugin should return
+a predefined error-type.
+
+In cases where the retry-ability of an error cannot be generally determined by
+the plugin, i.e. where an error code indicates a fatal, non-recoverable error
+in one case but a non-fatal, recoverable error in another case, the plugin might
+add configuration settings to let the user configure that property.
+
+## Configuration Options and Behaviors
+
+Telegraf must introduce a unified `startup_error_behavior` configuration option
+for inputs and output plugins which is handled directly by the Telegraf
+infrastructure and is not passed down to the plugins. The setting must be
+available on a per-plugin basis and defines how Telegraf behaves on startup
+errors.
+
+For all listed behaviors Telegraf might retry to startup the plugin for a
+limited number of times during the startup phase before actually processing
+data.
```

Comment:
What's the goal of this sentence?

Reply:
The goal is to include the current behavior! :-) Currently, Telegraf tries to start the plugin 3 times with an interval of 15 seconds before any of the configured behaviors apply. So if you set `retry` or `ignore` and expect Telegraf to startup immediately, you would be surprised. ;-)

Reply:
Tried to make this more clear in the latest commit. Let me know if this is what you intended!